src/stactools/goes/enums.py

```diff
@@ -0,0 +1,14 @@
+import enum
+from enum import Enum
+
+
+class ImageType(Enum):
+    FULL_DISK = enum.auto()
+    CONUS = enum.auto()
+    MESOSCALE = enum.auto()
+
+
+class Mode(Enum):
+    ABI_MODE_3 = 3
+    ABI_MODE_4 = 4
+    ABI_MODE_6 = 6
```

Comment:
I think these should be string Enums and set into the properties with string values parsed from the possible values. E.g. from the tests it looks like "ABI Mode 6" etc for mode and `Mesoscale` etc for the ImageType

I believe `goes:mesoscale-image-number` is also a restricted value, so should use an enum?